frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx

```diff
@@ -0,0 +1,162 @@
+  );
+  const [pvcResources, pvcLoaded, pvcLoadError] = useK8sGet<ListKind<PersistentVolumeClaimKind>>(
+    PersistentVolumeClaimModel,
+  );
+  const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
+
+  React.useEffect(() => {
+    // restrict pool management for default pool and external cluster
+    cephCluster?.metadata.name === CEPH_EXTERNAL_CR_NAME ||
+    blockPoolConfig?.metadata?.ownerReferences?.find(
+      (ownerReference) => ownerReference.name === OCS_INTERNAL_CR_NAME,
+    )
+      ? dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.NOTALLOWED })
+      : dispatch({
+          type: BlockPoolActionType.SET_POOL_NAME,
+          payload: poolName,
+        });
+  }, [blockPoolConfig, cephCluster, isLoaded, loadError, poolName]);
+
+  React.useEffect(() => {
+    if (scLoaded && pvcLoaded && state.poolStatus !== POOL_PROGRESS.NOTALLOWED) {
+      const poolScNames: string[] = scResources.items?.reduce((scList, sc) => {
+        if (sc.parameters?.pool === poolName) scList.push(sc.metadata?.name);
+        return scList;
+      }, []);
+      const pvcScNames: string[] = pvcResources.items?.map(getStorageClassName);
+
+      // intersection of scNames and pvcScNames
+      const usedScNames = poolScNames.filter((scName) => pvcScNames.includes(scName));
+
+      if (usedScNames.length) {
+        dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.BOUNDED });
+        setScNames(usedScNames);
+      }
+    }
+  }, [scResources, scLoaded, pvcResources, pvcLoaded, state.poolStatus, poolName]);
+
+  // Delete block pool
+  const onClick = () => {
+    handlePromise(k8sKill(CephBlockPoolModel, blockPoolConfig), () => close());
+  };
+
+  const pvcLink = () => {
+    history.push('/k8s/all-namespaces/persistentvolumeclaims');
+    close();
+  };
+
+  const MODAL_TITLE = t('ceph-storage-plugin~Delete Block Pool');
+
+  return (
+    <div className="modal-content modal-content--no-inner-scroll">
+      <ModalTitle close={close}>
+        <YellowExclamationTriangleIcon className="co-icon-space-r" /> {MODAL_TITLE}
+      </ModalTitle>
+      {isLoaded && pvcLoaded && scLoaded && !(loadError && pvcLoadError && scLoadError) ? (
+        <>
+          <ModalBody>
+            {state.poolStatus === POOL_PROGRESS.NOTALLOWED ? (
+              <div key="progress-modal">
+                <BlockPoolStatus
+                  status={state.poolStatus}
+                  name={state.poolName}
+                  error={state.errorMessage}
+                />
+              </div>
+            ) : state.poolStatus === POOL_PROGRESS.BOUNDED ? (
+              <Trans t={t} ns="ceph-storage-plugin">
+                <p>
+                  <strong>{{ poolName }}</strong> cannot be deleted. When a pool is bounded to PVC
+                  it cannot be deleted. Please detach all the resources from storage class(es){' '}
+                  <strong>{{ scNames }}</strong>.{' '}
```

Comment:
dont you want to destructure all array elements separated by `comma` and the last element by `and`?

Reply:
i accidentally removed the change, let me check how to add and for last element

Reply:
![image](https://user-images.githubusercontent.com/15073590/113859991-2596f880-97c3-11eb-89ee-8ca8a4027046.png)